### DISCUSSION TITLE
Help needed with conditional formatting for the TOC items of grouped table ? and change the TOC items color dynamically ?

### DISCUSSION THREAD

**Discussion by raviprsd18** [eclipse-birt/birt#1897] [Q&A] [answered]
Hello BIRT community,

Following is my problem statement. Any help will be highly appreciated.

1. Say I have a dataset that have studentId, studentName, studentGender and studentExamResult
   **| studentId | studentName | studentGener | studentExamResult |**
      | 1 | John | Male | Pass |
      | 2 | Albert | Male | Pass |
      | 3 | Katrin | Female | Fail |

2. I have designed my report as following.
    a. Insert the table => bind the dataset present in the point 1
    b. Group the table based on the StudentId
    c. Add the studentName as TOC for the grouped table based on the StudentId ( point b )
    
3. Is there any way to change the TOC items font color based on the studentExamResult column value.
   If studentExamResult is Pass => TOC item should be green color
   If studentExamResult is Fail => TOC item should be red color

I have attached the rptDesign based on the above scenario, I am able to loop till the table Group but not able to get the TOC items and change the color based on the above conditions, any help will be highly appreciated.

**Expected result screenshot**
![image](https://github.com/user-attachments/assets/fb5a32ca-f1fd-4960-9275-64a3a0bee7e1)

**RptDesign**
[TocFontExploration.zip](https://github.com/user-attachments/files/16901899/TocFontExploration.zip)

**Comment by raviprsd18** [comment]
@speckyspooky  or @hvbtup can you guys help here ? I am looking forward for your help.

**Comment by speckyspooky** [comment, marked as answer]
I take a look into this idea but cuurently there is no support to handle different styles according to a group value.
So only one color/style can be assigned and there is no scripting option too.

**Reply by raviprsd18** [reply]
@speckyspooky thanks for the reply.
